### PR TITLE
lakeFS Hadoop Filesystem to support Hadoop CLI

### LINKED ISSUE #2629: Check lakeFS Hadoop Filesystem with Hadoop CLI
There seems to be some missing functionality in the filesystem to support hadoop cli. A common error between `cp` and `mv` commands:

```
hadoop fs \
> -D fs.lakefs.access.key=<key> \
> -D fs.lakefs.secret.key=<secret> \
> -D fs.lakefs.endpoint=<endpoint>/api/v1 \
> -D fs.lakefs.impl=io.lakefs.LakeFSFileSystem \
> -libjars hadoop-lakefs-assembly-0.1.0.jar \
> -cp lakefs://example-repo/main/user2/yaaqov   lakefs://example-repo/main/user2/yaaqov2
cp: `tal-test': File exists
```

### PR DESCRIPTION
**Pull request by itaiad200**
Fixes #2629

### CODE REVIEW

File: clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java [hunk 1]
```diff
@@ -759,10 +759,22 @@ public String getScheme() {
...
     @Override
     public FileStatus[] globStatus(Path pathPattern) throws IOException {
-        // TODO: implement this method as part of https://github.com/treeverse/lakeFS/issues/2629
-        FileStatus fStatus = new FileStatus(0, false, 1, 20, 1,
-                new Path("globStatus-test-path"));
-        return new FileStatus[]{fStatus};
+        OPERATIONS_LOG.trace("globStatus({})", pathPattern);
+
+        FileStatus[] statuses;
+        try{
```
Comment (arielshaqed) [inline clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java:765]: nit:
```suggestion
        try {
```

**Review by arielshaqed — CHANGES_REQUESTED** [review]
1. How was this tested?

1. Does it would with [`count`](https://hadoop.apache.org/docs/r2.4.1/hadoop-project-dist/hadoop-common/FileSystemShell.html#count)?

1. Does the Hadoop FS spec test include something to test for this?  If so then we should ensure that it runs.

1. I don't understand how this implements [`globStatus`](https://hadoop.apache.org/docs/r2.7.1/api/org/apache/hadoop/fs/FileSystem.html#globStatus(org.apache.hadoop.fs.Path)) (and indeed IIUC the comment on the PR says that it does not).  If we provide only partial support, then I guess we would need to throw some "not implemented" exception when `path` contains any wildcards, rather than return an incorrect answer.

**Review by nopcoder — COMMENTED** [review]
Looking at 2.7.x s3a implementation - https://github.com/apache/hadoop/blob/release-2.7.1/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java it doesn't override this method.
Looking at the base implementation https://github.com/apache/hadoop/blob/release-2.7.1/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java#L1654
It uses a Globber implementation https://github.com/apache/hadoop/blob/release-2.7.1/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/Globber.java that calls the fs implementation to support the functionality.
Can you identify why the base implementation doesn't work for us? maybe there is a missing API or missed returning some data that will enable us this functionality?

File: clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java (hunk 1 above)
Comment (itaiad200) [reply clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java]: Code deleted.

**Comment by itaiad200** [comment]
Following @nopcoder & @arielshaqed advices, I checked again the base implementation.

What I assumed didn't work before, is actually in parity with the s3a behaviour. Therefore implementing the `globPattern` method is not needed and it's ok to rely on the basic implementation that will make the appropriate FS calls.

Was tested using FileSystemTracer that compares the behaviour of s3a and lakeFS FS with the Hadoop CLI which was the tool that initiated this bug (no other tool is calling the `globPattern`).

**Review by arielshaqed — APPROVED** [review]
This looks exactly right, only it remains untested.  I am approving as it is clearly better than what we have.  But I would be **really** happy if you could find & enable some test in the spec, and/or add a test of our own.

Thanks!

**Pull request merged**